front/app/containers/Admin/users/UserTableRow.tsx

```diff
@@ -100,20 +117,33 @@ const UserTableRow = ({
     }
   };
 
+  const changeRoleHandler = (changeToNormalUser: boolean) => {
+    setIsChangingToNormalUser(changeToNormalUser);
+
+    // We are showing the modal when setting to a normal user and for admins in i1 and for i2 when admin seats are being exceeded
+    const shouldOpenConfirmationInModal =
+      changeToNormalUser ||
+      !hasSeatBasedBillingEnabled ||
+      hasReachedOrIsOverLimit;
+    if (shouldOpenConfirmationInModal) {
+      setShowModal(true);
+      return;
+    }
+    changeRoles(user, changeToNormalUser);
```

Comment:
Could you still add a comment above this changeRoles line to add some context for future readers of the code?